server/controllers/categorised/employees.js

```diff
@@ -21,58 +51,208 @@ exports.list = function (req, res, next) {
+  .then(function (rows) {
+    if (!rows.length) {
+      throw codes.ERR_NOT_FOUND;
+    }
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+};
+
+/**
+* Update details of an employee referenced by an `id` in the database
+*
+* @param {object} request The express request object
+* @param {object} response The express response object
+* @param {object} next The express middleware next object
+*
+* @example
+* // PUT /employees/:id : Update details of an employee
+* var employees = require('categorised/employees');
+* employees.update(request, response, next);
+*/
+exports.update = function (req, res, next) {
+  var sql = 'UPDATE employee SET ? WHERE employee.id = ?';
+
+  db.exec(sql, [req.body, req.params.id])
+  .then(function (row) {
+    if (!row.affectedRows) { throw codes.ERR_NOT_FOUND; }
+
+    var sql2 =
+    'SELECT ' +
+    'employee.id, employee.code AS code_employee, employee.prenom, employee.name, ' +
+    'employee.postnom, employee.sexe, employee.dob, employee.date_embauche, employee.service_id, ' +
+    'employee.nb_spouse, employee.nb_enfant, employee.grade_id, employee.locked, grade.text, grade.basic_salary, ' +
+    'fonction.id AS fonction_id, fonction.fonction_txt, service.name AS service_txt, ' +
+    'employee.phone, employee.email, employee.adresse, employee.bank, employee.bank_account, employee.daily_salary, employee.location_id, ' +
+    'grade.code AS code_grade, debitor.uuid as debitor_uuid, debitor.text AS debitor_text,debitor.group_uuid as debitor_group_uuid, ' +
+    'creditor.uuid as creditor_uuid, creditor.text AS creditor_text, creditor.group_uuid as creditor_group_uuid, creditor_group.account_id ' +
+    'FROM employee ' +
+    ' JOIN grade ON employee.grade_id = grade.uuid ' +
+    ' JOIN fonction ON employee.fonction_id = fonction.id ' +
+    ' JOIN debitor ON employee.debitor_uuid = debitor.uuid ' +
+    ' JOIN creditor ON employee.creditor_uuid = creditor.uuid ' +
+    ' JOIN creditor_group ON creditor_group.uuid = creditor.group_uuid ' +
+    ' LEFT JOIN service ON service.id = employee.service_id ' +
+    'WHERE employee.id = ? ';
+
+    return db.exec(sql2, [req.params.id]);
+  })
+  .then(function (rows) {
+    if (!rows.length) {
+      throw codes.ERR_NOT_FOUND;
+    }
+    // returing date in 'yyyy-mm-dd' format
+    rows[0].dob = util.toMysqlDate(rows[0].dob);
+    rows[0].date_embauche = util.toMysqlDate(rows[0].date_embauche);
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+};
+
+/**
+* This function is responsible for creating a new employee in the database
+*
+* @param {object} request The express request object
+* @param {object} response The express response object
+* @param {object} next The express middleware next object
+*
+* @example
+* // POST /employees/ : Create a new employee
+* var employees = require('categorised/employees');
+* employees.create(request, response, next);
+*/
+exports.create = function (req, res, next) {
+  var sql =
```

Comment:
This SQL would be safer and more flexible if you used the "INSERT INTO table SET key = value" syntax.  It would look  like this:
```js
var sql = 'INSERT INTO employee SET ?;';
db.exec(sql, req.body)
.then(function () { /* handle success, etc */ });
```
**NOTE** If you choose to use this syntax, you should make sure that `req.body` contains all required properties of the database table.  I usually use a for-loop to check them.  For example:
```js
var requiredKeys = ['code', 'prenom', 'dob'];
var employee = req.body;

// ensure the client sent all required fields
var validRequest = requiredKeys.every(function (key) {
  return employee[key] !== null && employee[key] !== undefined;
});

if (!validRequest) {
  return res.status(400).json({
    code : 'ERR_MISSING_PROPERTIES',
    reason : 'The employee creation path requires a code, name and date of birth ' +
      'please make sure these all exist and re-submit.'
   });
}

// create an employee as usual ...
```